service/service_core.go

```diff
@@ -48,6 +48,7 @@ type metricsCore struct {
 	statsReporter    tally.CachedStatsReporter
 	metricsCloser    io.Closer
 	runtimeCollector *metrics.RuntimeCollector
+	versionEmitter   versionMetricsEmitter
```

Comment:
fair enough. seems unnecessary given that there's basically zero data in it, and now the thing can be nil